src/mlpack/methods/hdbscan/hdbscan.hpp

```diff
@@ -0,0 +1,218 @@
+/**
+ * @file hdbscan.hpp
+ * @author Sudhanshu Ranjan
+ *
+ * An implementation of the HDBSCAN clustering method.
+ * HDBSCAN works with variable density.
+ *  
+ *
+ * @code
+ * @inproceedings{
+ *   author = {R. J. G. B. Campello, D. Moulavi & Sander, J.},
+ *   title = {{Density-Based Clustering Based
+ *             on Hierarchical Density Estimates.}},
+ * }
+ * @endcode
+ * Reference : "https://github.com/EdwardRaff/JSAT/blob/master/JSAT/
+ *              src/jsat/clustering/HDBSCAN.java"
+ * Reference : "http://hdbscan.readthedocs.io/en/latest/how_hdbscan_works.html"
+ */
+#ifndef __MLPACK_METHODS_HDBSCAN_HDBSCAN_HPP
+#define __MLPACK_METHODS_HDBSCAN_HDBSCAN_HPP
+
+#include <mlpack/core.hpp>
+#include <boost/dynamic_bitset.hpp>
+#include <mlpack/methods/neighbor_search/neighbor_search.hpp>
+#include <mlpack/methods/emst/dtb.hpp>
+#include <mlpack/core/metrics/hdbscan_metric.hpp>
+
+namespace mlpack {
+namespace hdbscan {
+
+template<typename NeighborSearch = neighbor::NeighborSearch
```

Comment:
Could you please add some documentation discussing what the class does?  This can be similar to the comment at the top of the file, but it should also detail each of the template parameters and what they do.